tests/postgres_tests/test_aggregates.py

```diff
@@ -302,6 +302,54 @@ def test_array_agg_filter_slice(self):
         )
         self.assertCountEqual(qs, [[], [5]])
 
+    def test_array_agg_with_default_empty_list(self):
+        aggr1 = AggregateTestModel.objects.create(integer_field=1)
+        aggr2 = AggregateTestModel.objects.create(integer_field=2)
+        StatTestModel.objects.bulk_create([
+            StatTestModel(related_field=aggr1, int1=1, int2=0),
+            StatTestModel(related_field=aggr1, int1=2, int2=1),
+            StatTestModel(related_field=aggr2, int1=3, int2=0),
+            StatTestModel(related_field=aggr2, int1=4, int2=1),
+            StatTestModel(related_field=aggr2, int1=5, int2=0),
+        ])
+
+        annotated_queryset = AggregateTestModel.objects.filter(
+            pk__in=[aggr1.pk, aggr2.pk]).annotate(
+            array=ArrayAgg("stattestmodel__int1", filter=Q(stattestmodel__int2=3))
+        )
+
+        result = annotated_queryset.aggregate(
+            aggregated_results=ArrayAgg('array', default=Value([]))
+        )
+
+        expected_result = {'aggregated_results': [[], []]}
+        self.assertEqual(result, expected_result)
+
+    def test_array_agg_with_empty_filter_value(self):
+        aggr1 = AggregateTestModel.objects.create(integer_field=1)
+        aggr2 = AggregateTestModel.objects.create(integer_field=2)
+        StatTestModel.objects.bulk_create([
+            StatTestModel(related_field=aggr1, int1=1, int2=0),
+            StatTestModel(related_field=aggr1, int1=2, int2=1),
+            StatTestModel(related_field=aggr2, int1=3, int2=0),
+            StatTestModel(related_field=aggr2, int1=4, int2=1),
+            StatTestModel(related_field=aggr2, int1=5, int2=0),
+        ])
+
+        annotated_queryset = AggregateTestModel.objects.filter(
+            pk__in=[aggr1.pk, aggr2.pk]).annotate(
+            array=ArrayAgg("stattestmodel__int1",
+                           filter=Q(stattestmodel__int2__in=[]))
+
+        )
+
+        result = annotated_queryset.aggregate(
+            aggregated_results=ArrayAgg('array', default=Value([]))
+        )
```

Comment:
Nice work in getting a failing test happening especially since this is your first time 🏆  Definitely on the right track.

My only suggestion would be to remove the test data since the test isn't testing for that - it's testing for default types when an `EmptyResultSet` is raised. Ideally tests should be as concise as possible to focus on the problem and make them easier to read :)

This is an equivalent failing test:

```
    def test_array_agg_with_empty_filter_value(self):
        result = AggregateTestModel.objects.annotate(
            array=ArrayAgg("stattestmodel__int1", filter=Q(pk__in=[]))
        ).aggregate(aggregated_results=ArrayAgg("array", default=Value([])))
        expected_result = {"aggregated_results": [[], [], [], []]}
        self.assertEqual(result, expected_result)
```

Note the filter is simply `pk__in=[]` which should be all that's needed to trigger an `EmptyResultSet`.

Reply:
Or maybe this one is better, removed the second `ArrayAgg` 🤔 

```
    def test_array_agg_with_empty_filter_value(self):
        result = (
            AggregateTestModel.objects.annotate(
                array=ArrayAgg(
                    "stattestmodel__int1", filter=Q(pk__in=[]), default=Value([])
                )
            )
            .values_list("array", flat=True)
            .order_by("pk")
        )
        expected_result = [[], [], [], []]
        self.assertQuerySetEqual(result, expected_result)
```

Reply:
Thanks for confirming I'm looking in the right places :) being new I referenced an existing test and tried to write a few learning tests of my own, lots to improve upon certainly as you pointed out. 

Saw a lot happened while I was having dinner in the ticket as well, let me take everything onboard and try again.